rpcserver.go

```diff
@@ -3058,17 +3063,32 @@ func marshalLeafKey(leafKey universe.LeafKey) *unirpc.AssetKey {
 // Taproot Asset commitment, and script_key is the script_key of the asset
 // within the Taproot Asset commitment for the given asset_id or group_key.
 func (r *rpcServer) AssetLeafKeys(ctx context.Context,
-	req *unirpc.ID) (*unirpc.AssetLeafKeyResponse, error) {
+	req *unirpc.AssetLeafKeysRequest) (*unirpc.AssetLeafKeyResponse, error) {
 
-	universeID, err := UnmarshalUniID(req)
+	if req == nil {
+		return nil, fmt.Errorf("request must be set")
+	}
+
+	universeID, err := UnmarshalUniID(req.Id)
 	if err != nil {
 		return nil, err
 	}
 
 	// TODO(roasbeef): tell above if was tring or not, then would set
 	// below diff
 
-	leafKeys, err := r.cfg.UniverseArchive.UniverseLeafKeys(ctx, universeID)
+	if req.Limit > universe.MaxPageSize || req.Limit < 0 {
+		return nil, fmt.Errorf("invalid request limit")
```

Comment:
nit: include max page size in error message.

Reply:
oh that was my previous message, thought of leaving it out in order to not expose server params